std/format.d

```diff
@@ -3040,19 +3040,45 @@ if (isInputRange!T)
             static if (is(StringTypeOf!T))
             {
                 auto s = val[0 .. f.precision < $ ? f.precision : $];
+
+                size_t width;
+                if (f.width > 0)
+                {
+                    // strings that are fully made of ASCII characters
+                    // can be aligned w/o graphemeStride
+                    bool onlyAscii = true;
```

Comment:
```d
bool onlyAscii = s.byCodeUnit.all!isASCII;
```

If only. I understand why you broke it out into a custom loop; it's such a shame we have to work around DMD's inliner all the time.

Reply:
Yes, indeed. I thought to use a higher order function but since testing for "ascii-ness" is trivial i choose to avoid imports to std.algo + std.scii.

Reply:
IMO one of these days we should spearhead a dmd optimizer improvement project, where we identify all such instances of deficiency in the dmd inliner and push for them to be fixed.  I'm thinking that all range-based idioms really need stronger support from D's reference compiler, since they're supposed to be one of D's selling points. More aggressive inlining is part of the necessary optimization in order to make range-based code more performant.

The other big area that will probably make a big impact is loop unrolling, which IIRC Walter says isn't done by dmd except in special cases. But that is probably far more complex than improving the inliner.

Reply:
@BBasile do you remember how big the performance penalty of using `s.byCodeUnit.all!isASCII` was?